modules/reactor/src/meshgenerators/TriPinHexAssemblyGenerator.C

```diff
@@ -0,0 +1,507 @@
+            secondary_side_length_1 * secondary_side_length_1) /
+           2.0 / primary_side_length_1 / side_length) /
+      M_PI * 180.0);
+
+  std::vector<Real> azimuthal_tangent_0;
+  std::vector<Real> azimuthal_tangent_1;
+  std::vector<Real> azimuthal_list;
+  for (unsigned int i = 0; i < num_sectors_per_side; i++)
+  {
+    azimuthal_list.push_back(
+        atan((Real)i * side_length / num_sectors_per_side * sin(alpha_angle_0 / 180.0 * M_PI) /
+             (primary_side_length_0 -
+              (Real)i * side_length / num_sectors_per_side * cos(alpha_angle_0 / 180.0 * M_PI))) /
+        M_PI * 180.0);
+  }
+  for (unsigned int i = 0; i < num_sectors_per_side; i++)
+  {
+    azimuthal_list.push_back(
+        azimuthal_angle_0 +
+        atan((Real)i * side_length / num_sectors_per_side * sin(alpha_angle_1 / 180.0 * M_PI) /
+             (primary_side_length_1 -
+              (Real)i * side_length / num_sectors_per_side * cos(alpha_angle_1 / 180.0 * M_PI))) /
+            M_PI * 180.0);
+  }
+  for (unsigned int i = 0; i < num_sectors_per_side * 2; i++)
+  {
+    azimuthal_list.push_back(azimuthal_list[i] + 180.0);
+  }
+
+  std::vector<Real> ring_radii_corr;
+  if (has_rings)
+  {
+    if (preserve_volumes)
+    {
+      const Real corr_factor = radiusCorrectionFactor(azimuthal_list);
+      for (unsigned int i = 0; i < ring_radii.size(); i++)
+        ring_radii_corr.push_back(ring_radii[i] * corr_factor);
+    }
+    else
+      ring_radii_corr = ring_radii;
+    if (ring_radii_corr.back() > (side_length / 2.0 - std::abs(ring_offset)) * std::sqrt(3.0) / 2.0)
+      paramError("ring_radii",
+                 "The radii of the rings cannot exceed the boundary of the diamond section.");
+  }
+
+  // build the first slice of the polygon.
+  auto mesh0 = buildGeneralSlice(
+      ring_radii_corr,
+      ring_intervals,
+      std::vector<Real>(ring_radii_corr.size(), 1.0),
+      {std::vector<Real>(ring_radii_corr.size(), 0.0),
+       std::vector<Real>(ring_radii_corr.size(), 0.0),
+       std::vector<unsigned int>(ring_radii_corr.size(), 0),
+       std::vector<Real>(ring_radii_corr.size(), 1.0)},
+      {std::vector<Real>(ring_radii_corr.size(), 0.0),
+       std::vector<Real>(ring_radii_corr.size(), 0.0),
+       std::vector<unsigned int>(ring_radii_corr.size(), 0),
+       std::vector<Real>(ring_radii_corr.size(), 1.0)},
+      std::vector<Real>(),
+      std::vector<unsigned int>(),
+      std::vector<Real>(),
+      {std::vector<Real>(), std::vector<Real>(), std::vector<unsigned int>(), std::vector<Real>()},
+      {std::vector<Real>(), std::vector<Real>(), std::vector<unsigned int>(), std::vector<Real>()},
+      primary_side_length_0,
+      secondary_side_length_0,
+      num_sectors_per_side,
+      background_intervals,
+      1.0,
+      {0.0, 0.0, 0, 1.0},
+      {0.0, 0.0, 0, 1.0},
+      node_id_background_meta,
+      azimuthal_angle_0,
+      azimuthal_tangent_0,
+      /* side_index = */ 1,
+      false,
```

Comment:
umm should this be true?
the pictured mesh looks real skewed in the example, this would help right?

Reply:
Currently it has to be triangular elements in the center. I may enable quad mesh option, which would need some algorithm to make sure the stitching can be done, in a future PR.